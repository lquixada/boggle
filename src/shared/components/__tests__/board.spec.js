import React from 'react'
import { mount } from 'enzyme'
import { fromJS } from 'immutable'

import { Board } from '..'

describe('<Board />', () => {
  it('is empty by default', () => {
    const matrix = fromJS([
      [' ', ' ', ' ', ' '],
      [' ', ' ', ' ', ' '],
      [' ', ' ', ' ', ' '],
      [' ', ' ', ' ', ' ']
    ])
    const component = mount(<Board matrix={matrix} />)

    expect(component).toMatchSnapshot()
  })

  it('is filled when the game has started', () => {
    const matrix = fromJS([
      ['E', 'H', 'R', 'T'],
      ['B', 'I', 'T', 'E'],
      ['G', 'I', 'S', 'F'],
      ['A', 'M', 'N', 'O']
    ])
    const component = mount(<Board matrix={matrix} />)

    expect(component).toMatchSnapshot()
  })
})
