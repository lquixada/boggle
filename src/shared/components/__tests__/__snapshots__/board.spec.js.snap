// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Board /> is empty by default 1`] = `
<Board
  matrix={
    Immutable.List [
      Immutable.List [
        " ",
        " ",
        " ",
        " ",
      ],
      Immutable.List [
        " ",
        " ",
        " ",
        " ",
      ],
      Immutable.List [
        " ",
        " ",
        " ",
        " ",
      ],
      Immutable.List [
        " ",
        " ",
        " ",
        " ",
      ],
    ]
  }
>
  <styled.div>
    <div
      className="sc-bdVaJa iOMCKQ"
    >
      <styled.table>
        <table
          className="sc-bwzfXH jwfMIB"
        >
          <tbody>
            <Row
              key="0"
              letters={
                Immutable.List [
                  " ",
                  " ",
                  " ",
                  " ",
                ]
              }
            >
              <tr>
                <styled.td
                  key="0"
                >
                  <td
                    className="sc-htpNat kcMuME"
                  >
                     
                  </td>
                </styled.td>
                <styled.td
                  key="1"
                >
                  <td
                    className="sc-htpNat kcMuME"
                  >
                     
                  </td>
                </styled.td>
                <styled.td
                  key="2"
                >
                  <td
                    className="sc-htpNat kcMuME"
                  >
                     
                  </td>
                </styled.td>
                <styled.td
                  key="3"
                >
                  <td
                    className="sc-htpNat kcMuME"
                  >
                     
                  </td>
                </styled.td>
              </tr>
            </Row>
            <Row
              key="1"
              letters={
                Immutable.List [
                  " ",
                  " ",
                  " ",
                  " ",
                ]
              }
            >
              <tr>
                <styled.td
                  key="0"
                >
                  <td
                    className="sc-htpNat kcMuME"
                  >
                     
                  </td>
                </styled.td>
                <styled.td
                  key="1"
                >
                  <td
                    className="sc-htpNat kcMuME"
                  >
                     
                  </td>
                </styled.td>
                <styled.td
                  key="2"
                >
                  <td
                    className="sc-htpNat kcMuME"
                  >
                     
                  </td>
                </styled.td>
                <styled.td
                  key="3"
                >
                  <td
                    className="sc-htpNat kcMuME"
                  >
                     
                  </td>
                </styled.td>
              </tr>
            </Row>
            <Row
              key="2"
              letters={
                Immutable.List [
                  " ",
                  " ",
                  " ",
                  " ",
                ]
              }
            >
              <tr>
                <styled.td
                  key="0"
                >
                  <td
                    className="sc-htpNat kcMuME"
                  >
                     
                  </td>
                </styled.td>
                <styled.td
                  key="1"
                >
                  <td
                    className="sc-htpNat kcMuME"
                  >
                     
                  </td>
                </styled.td>
                <styled.td
                  key="2"
                >
                  <td
                    className="sc-htpNat kcMuME"
                  >
                     
                  </td>
                </styled.td>
                <styled.td
                  key="3"
                >
                  <td
                    className="sc-htpNat kcMuME"
                  >
                     
                  </td>
                </styled.td>
              </tr>
            </Row>
            <Row
              key="3"
              letters={
                Immutable.List [
                  " ",
                  " ",
                  " ",
                  " ",
                ]
              }
            >
              <tr>
                <styled.td
                  key="0"
                >
                  <td
                    className="sc-htpNat kcMuME"
                  >
                     
                  </td>
                </styled.td>
                <styled.td
                  key="1"
                >
                  <td
                    className="sc-htpNat kcMuME"
                  >
                     
                  </td>
                </styled.td>
                <styled.td
                  key="2"
                >
                  <td
                    className="sc-htpNat kcMuME"
                  >
                     
                  </td>
                </styled.td>
                <styled.td
                  key="3"
                >
                  <td
                    className="sc-htpNat kcMuME"
                  >
                     
                  </td>
                </styled.td>
              </tr>
            </Row>
          </tbody>
        </table>
      </styled.table>
    </div>
  </styled.div>
</Board>
`;

exports[`<Board /> is filled when the game has started 1`] = `
<Board
  matrix={
    Immutable.List [
      Immutable.List [
        "E",
        "H",
        "R",
        "T",
      ],
      Immutable.List [
        "B",
        "I",
        "T",
        "E",
      ],
      Immutable.List [
        "G",
        "I",
        "S",
        "F",
      ],
      Immutable.List [
        "A",
        "M",
        "N",
        "O",
      ],
    ]
  }
>
  <styled.div>
    <div
      className="sc-bdVaJa iOMCKQ"
    >
      <styled.table>
        <table
          className="sc-bwzfXH jwfMIB"
        >
          <tbody>
            <Row
              key="0"
              letters={
                Immutable.List [
                  "E",
                  "H",
                  "R",
                  "T",
                ]
              }
            >
              <tr>
                <styled.td
                  key="0"
                >
                  <td
                    className="sc-htpNat kcMuME"
                  >
                    E
                  </td>
                </styled.td>
                <styled.td
                  key="1"
                >
                  <td
                    className="sc-htpNat kcMuME"
                  >
                    H
                  </td>
                </styled.td>
                <styled.td
                  key="2"
                >
                  <td
                    className="sc-htpNat kcMuME"
                  >
                    R
                  </td>
                </styled.td>
                <styled.td
                  key="3"
                >
                  <td
                    className="sc-htpNat kcMuME"
                  >
                    T
                  </td>
                </styled.td>
              </tr>
            </Row>
            <Row
              key="1"
              letters={
                Immutable.List [
                  "B",
                  "I",
                  "T",
                  "E",
                ]
              }
            >
              <tr>
                <styled.td
                  key="0"
                >
                  <td
                    className="sc-htpNat kcMuME"
                  >
                    B
                  </td>
                </styled.td>
                <styled.td
                  key="1"
                >
                  <td
                    className="sc-htpNat kcMuME"
                  >
                    I
                  </td>
                </styled.td>
                <styled.td
                  key="2"
                >
                  <td
                    className="sc-htpNat kcMuME"
                  >
                    T
                  </td>
                </styled.td>
                <styled.td
                  key="3"
                >
                  <td
                    className="sc-htpNat kcMuME"
                  >
                    E
                  </td>
                </styled.td>
              </tr>
            </Row>
            <Row
              key="2"
              letters={
                Immutable.List [
                  "G",
                  "I",
                  "S",
                  "F",
                ]
              }
            >
              <tr>
                <styled.td
                  key="0"
                >
                  <td
                    className="sc-htpNat kcMuME"
                  >
                    G
                  </td>
                </styled.td>
                <styled.td
                  key="1"
                >
                  <td
                    className="sc-htpNat kcMuME"
                  >
                    I
                  </td>
                </styled.td>
                <styled.td
                  key="2"
                >
                  <td
                    className="sc-htpNat kcMuME"
                  >
                    S
                  </td>
                </styled.td>
                <styled.td
                  key="3"
                >
                  <td
                    className="sc-htpNat kcMuME"
                  >
                    F
                  </td>
                </styled.td>
              </tr>
            </Row>
            <Row
              key="3"
              letters={
                Immutable.List [
                  "A",
                  "M",
                  "N",
                  "O",
                ]
              }
            >
              <tr>
                <styled.td
                  key="0"
                >
                  <td
                    className="sc-htpNat kcMuME"
                  >
                    A
                  </td>
                </styled.td>
                <styled.td
                  key="1"
                >
                  <td
                    className="sc-htpNat kcMuME"
                  >
                    M
                  </td>
                </styled.td>
                <styled.td
                  key="2"
                >
                  <td
                    className="sc-htpNat kcMuME"
                  >
                    N
                  </td>
                </styled.td>
                <styled.td
                  key="3"
                >
                  <td
                    className="sc-htpNat kcMuME"
                  >
                    O
                  </td>
                </styled.td>
              </tr>
            </Row>
          </tbody>
        </table>
      </styled.table>
    </div>
  </styled.div>
</Board>
`;
